Clarify docs about CA fingerprints and Fleet (#84056)
elastic/elasticsearch: docs/reference/setup/install/connect-clients.asciidoc

==== Connect clients to {es}

> When you start {es} for the first time, TLS is configured automatically for the

Author: @jkakavas Can you take another look? I thought it made sense to mention the file path where we talk about the file being stored on disk. 

Also...is hex-coded correct? Or should this be hex-encoded?

Reviewer: Good catch ! It’s hex-encoded

HTTP layer. A CA certificate is generated and stored on disk at

> `$ES_HOME/config/certs/http_ca.crt`. The hex-coded SHA-256 fingerprint of this

Author: ```suggestion
`$ES_HOME/config/certs/http_ca.crt`. The hex-encoded SHA-256 fingerprint of this
```

> `$ES_HOME/config/certs/http_ca.crt`. The hex-coded  SHA-256 fingerprint of this

Author: ```suggestion
`$ES_HOME/config/certs/http_ca.crt`. The hex-coded SHA-256 fingerprint of this
```

SHA-256 fingerprint of this certificate is also output to the terminal. Any 
clients that connect to {es}, such as the 
https://www.elastic.co/guide/en/elasticsearch/client/index.html[{es} Clients],
{beats}, standalone {agent}s, and {ls} must validate that they trust the
certificate that {es} uses for HTTPS. {fleet-server} and {fleet}-managed

> {agent}s are automatically configured to trust the fingerprint of the CA.

Reviewer: ```suggestion
{agent}s are automatically configured to trust the CA certificate.
```

The CA certificate is what is ultimately trusted. Whether the actual CA certificate  or the fingerprint of said CA certificate is used to do that, is more of an internal implementation detail for us. I think that expressing this as "trusting the certificate" matches terminology that users are used to read and is the least probable to raise any eyebrows.

Other clients can establish trust by using either the fingerprint of the CA
certificate or the CA certificate itself.

If the auto-configuration process already completed, you can still obtain the 

> fingerprint of the security certificate. You can also copy the CA certificate

Reviewer: I know this is not part of this change but we can also mention the path to the CA certificate  in these instructions which is always : `$ES_HOME/config/certs/http_ca.crt` and later on in this file, maybe in the top after the " You can also copy the CA certificate to your machine and configure your client to use it. " section ? 

Also, we can remove the 
`<timestamp>:: The timestamp of when the auto-configuration process created the security files directory.` callout as it is a remnant of a previous iteration.

Reviewer 2: @dedemorton Let me know how you want to handle this. I'm happy to do one of the following:

A) Perform this work in a separate PR once this is merged in.
B) Push these changes to your branch.
C) Let you make the changes on this branch.

I mostly don't want to block you or push stuff to your PR without your permission.
